e2e/test/iothub/properties/PropertiesE2ETests.cs

```diff
@@ -0,0 +1,483 @@
+
+            await deviceClient.SubscribeToWritablePropertiesEventAsync(null, null).ConfigureAwait(false);
+            await deviceClient.CloseAsync().ConfigureAwait(false);
+        }
+
+        private async Task Properties_ServiceSetsWritablePropertyAndDeviceReceivesItOnNextGetAsync(Client.TransportType transport)
+        {
+            var propName = Guid.NewGuid().ToString();
+            var propValue = Guid.NewGuid().ToString();
+
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, _devicePrefix).ConfigureAwait(false);
+            using var registryManager = RegistryManager.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+            using var deviceClient = DeviceClient.CreateFromConnectionString(testDevice.ConnectionString, transport);
+
+            var twinPatch = new Twin();
+            twinPatch.Properties.Desired[propName] = propValue;
+            await registryManager.UpdateTwinAsync(testDevice.Id, twinPatch, "*").ConfigureAwait(false);
+
+            ClientProperties deviceTwin = await deviceClient.GetClientPropertiesAsync().ConfigureAwait(false);
+            if (deviceTwin.Writable.TryGetValue(propName, out string propFromCollection))
+            {
+                Assert.AreEqual<string>(propFromCollection, propValue);
+            } else
+            {
+                Assert.Fail("Property not found in ClientProperties");
+            }
+            await deviceClient.CloseAsync().ConfigureAwait(false);
+            await registryManager.CloseAsync().ConfigureAwait(false);
+        }
+
+        private async Task Properties_DeviceSetsPropertyAndServiceReceivesItAsync(Client.TransportType transport)
+        {
+            var propName = Guid.NewGuid().ToString();
+            var propValue = Guid.NewGuid().ToString();
+
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, _devicePrefix).ConfigureAwait(false);
+            using var registryManager = RegistryManager.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+            using var deviceClient = DeviceClient.CreateFromConnectionString(testDevice.ConnectionString, transport);
+
+            var patch = new ClientPropertyCollection();
+            patch[propName] = propValue;
+            await deviceClient.UpdateClientPropertiesAsync(patch).ConfigureAwait(false);
+            await deviceClient.CloseAsync().ConfigureAwait(false);
+
+            Twin serviceTwin = await registryManager.GetTwinAsync(testDevice.Id).ConfigureAwait(false);
+            Assert.AreEqual<string>(serviceTwin.Properties.Reported[propName].ToString(), propValue);
+
+            Logger.Trace("verified " + serviceTwin.Properties.Reported[propName].ToString() + "=" + propValue);
+        }
+
+        private async Task Properties_ServiceDoesNotCreateNullPropertyInCollectionAsync(Client.TransportType transport)
```

Comment:
how about we also we add a test that verifies that updating an existing property patch with null value removes the property from the twin

Reply:
added